GaeManda/Projects/Presentation/GMDProfile/Implementations/UserProfileViewController.swift

```diff
@@ -17,10 +22,275 @@ final class UserProfileViewController:
+	}()
+	
+	private let collectionView: UICollectionView = {
+		let layout = UICollectionViewFlowLayout()
+		layout.scrollDirection = .horizontal
+		layout.minimumLineSpacing = 0
+		
+		let collectionView = UICollectionView(
+			frame: .zero,
+			collectionViewLayout: layout
+		)
+		collectionView.showsHorizontalScrollIndicator = false
+		collectionView.isPagingEnabled = true
+		collectionView.backgroundColor = .gray40
+		collectionView.register(
+			DogsCollectionViewCell.self,
+			forCellWithReuseIdentifier: DogsCollectionViewCell.idenfier
+		)
+		collectionView.layer.cornerRadius = 4
+		
+		return collectionView
+	}()
 	
 	override func viewDidLoad() {
 		super.viewDidLoad()
+		collectionView.delegate = self
+		setupUI()
+	}
+	
+	private func setupUI() {
+		view.backgroundColor = .white
+		title = "프로필"
+		indicatorView.indicatorCount = dogInformations.count
+		
+		DispatchQueue.main.async {
+			self.collectionView.scrollToItem(
+				at: IndexPath(item: 1, section: 0),
+				at: .right,
+				animated: false
+			)
+		}
+		
+		setNavigationTitleFont(.b20)
 		
-		view.backgroundColor = .red
+		setupSubviews()
+		setConstraints()
+		setNavigationBarButton()
+		bind()
+	}
+	
+	private func setNavigationBarButton() {
+		let notificationBarButton = UIBarButtonItem(customView: notificationButton)
+		let settingBarButton = UIBarButtonItem(customView: settingButton)
+		
+		navigationItem.rightBarButtonItems = [settingBarButton, notificationBarButton]
+	}
+	
+	private func setupSubviews() {
+		view.addSubview(nickNameLabel)
+		view.addSubview(profileEditButton)
+		view.addSubview(sexAndAgeLabel)
+		view.addSubview(profileImageView)
+		view.addSubview(indicatorView)
+		view.addSubview(collectionView)
+	}
+	
+	private func setConstraints() {
+		nickNameLabel.snp.makeConstraints { make in
+			make.centerX.equalToSuperview()
+			make.top.equalToSuperview().offset(160)
+		}
+		
+		profileEditButton.snp.makeConstraints { make in
+			make.leading.equalTo(nickNameLabel.snp.trailing).offset(12)
+			make.centerY.equalTo(nickNameLabel)
+			make.width.equalTo(36)
+			make.height.equalTo(16)
+		}
+		
+		sexAndAgeLabel.snp.makeConstraints { make in
+			make.centerX.equalToSuperview()
+			make.top.equalTo(nickNameLabel.snp.bottom).offset(12)
+		}
+		
+		profileImageView.snp.makeConstraints { make in
+			make.centerX.equalToSuperview()
+			make.top.equalTo(sexAndAgeLabel.snp.bottom).offset(12)
+			make.height.width.equalTo(160)
+		}
+		
+		indicatorView.snp.makeConstraints { make in
+			make.top.equalTo(profileImageView.snp.bottom).offset(32)
+			make.trailing.equalTo(collectionView)
+		}
+		
+		collectionView.snp.makeConstraints { make in
+			make.top.equalTo(indicatorView.snp.bottom).offset(8)
+			make.leading.equalToSuperview().offset(29)
+			make.trailing.equalToSuperview().offset(-29)
+			make.height.equalTo(102)
+		}
+	}
+	
+	private func bind() {
+		collectionViewBind()
+	}
+	
+	private func collectionViewBind() {
+		Observable.of(dogInformations)
```

Comment:
아래에서도 dogInformations를 사용하고 있어서 시퀀스가 중복해서 생겨요!~! 
혹시 share를 사용하거나 Driver로 바꾸는 것이 어떤가요 ~?
```suggestion
		let myDriver = Driver.of(dogInformations)
		myDriver
			.map { $0.count }
			.drive(with: self) { owner, count in
				if count == 1 {
					owner.collectionView.isScrollEnabled = false
				} else {
					owner.collectionView.isScrollEnabled = true
				}
			}
			.disposed(by: disposeBag)
		
		myDriver
			.map { item in
				guard let last = item.last, let first = item.first else { return item }
				
				var dogs = item
				dogs.insert(last, at: 0)
				dogs.append(first)
				
				return dogs
			}
			.drive(
				collectionView.rx.items(
					cellIdentifier: DogsCollectionViewCell.idenfier,
					cellType: DogsCollectionViewCell.self
				)
			) { _, item, cell in
				cell.configuration(item)
			}
			.disposed(by: disposeBag)
```